src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs

```diff
@@ -225,16 +226,20 @@ private void OnTimer(object sender, ElapsedEventArgs e)
                 adjustedInvocationTime = ScheduleStatus.Next;
             }
 
+            // Create the Last value with the adjustedInvocationTime; otherwise, the listener will
+            // consider this a schedule change when the host next starts.
             ScheduleStatus = new ScheduleStatus
             {
-                Last = invocationTime,
+                Last = adjustedInvocationTime,
                 Next = _schedule.GetNextOccurrence(adjustedInvocationTime)
             };
 
             if (ScheduleMonitor != null)
             {
                 await ScheduleMonitor.UpdateStatusAsync(_timerName, ScheduleStatus);
             }
+
+            _trace.Verbose($"Function '{_timerName}' updated status: Last='{ScheduleStatus.Last.ToString("o")}', Next='{ScheduleStatus.Next.ToString("o")}'");
```

Comment:
My intent was to log the 'interesting' scenarios to aid debugging. However, I think you're right that printing out the current status for non-monitor scenarios (which will typically be those with less than 1 minute timers) isn't really that interesting and will result in a lot of noise. We've seen most of the issues surrounding longer timers using the monitor. I'll move this inside the block above.